### ISSUE TITLE
Bordered cells in a table

### ISSUE THREAD

**Issue by limo49** [PyFPDF/fpdf2#45]
How can be created a table with borders to each cells?
Using write_html, I can set borders to the TABLE but not to each TD.

**Comment by Lucas-C** [comment]
Could you provide a short code snippet of how you create your table please?

**Comment by alexanderankin** [comment]
heres an minimum table

```python
class Thing(fpdf.FPDF, fpdf.HTMLMixin): pass;
f = Thing();
f.add_page();
f.write_html('<table><thead><tr><th width="100">left</th><th width="200">cent</th><th width="100">right</th></tr></thead><tbody><tr><td>1</td><td>2</td><td>3</td></tr><tr><td style="">4</td><td>5</td><td>6</td></tr></tbody></table>');
f.output('output.pdf', 'F');
```

just putting it here in case its useful for figuring out what to do - you can put these all on one command line (and put `os.system('evince output.pdf');` afterwards) and try new changes out.

**Comment by Lucas-C** [comment]
There is an example of HTML to enable `<table>` borders:

```html
<table border="1"><thead><tr>
    <th width="250">left</th><th width="500">center</th><th width="250">right</th>
</tr></thead><tbody><tr>
    <td>1</td><td>2</td><td>3</td>
</tr><tr>
    <td>4</td><td>5</td><td>6</td>
</tr></tbody></table>
```

I added tests and fixed a few bugs with `<table>` usage with `write_html` as part of this PR: https://github.com/PyFPDF/pyfpdf/pull/46

**Comment by limo49** [comment]
Thanks for the answers but your examples don't set borders around each single cell (in version 2.0.6)

Instead of:
![image](https://user-images.githubusercontent.com/29986135/104183626-8d32e700-5412-11eb-9118-b445c63e1be2.png)

I need:
![image](https://user-images.githubusercontent.com/29986135/104183742-b6ec0e00-5412-11eb-87b3-24194b747fd3.png)

**Comment by Lucas-C** [comment]
Yes, that's why I introduced `table_line_separators` on https://github.com/PyFPDF/pyfpdf/pull/46

Feel free to add your comments on it: does it: do you think this parameter naming is fine? Given the test output in `test/html/test_html_table_line_separators.pdf `, will that achieve what you need?

I think we can include that PR as part of version 2.2.0 that should be released soo.

**Comment by Lucas-C** [comment]
Note that for more advanced HTML to PDF conversion, you may want to check https://github.com/Kozea/WeasyPrint

**Comment by limo49** [comment]
It seems `table_line_separators` is OK.
Thanks

**Comment by Lucas-C** [comment]
Closing this then, as it has been released in v2.2.0: https://pypi.org/project/fpdf2/
